rotkehlchen/globaldb/handler.py

```diff
@@ -1203,6 +1208,144 @@ def get_historical_price_data(source: HistoricalPriceOracle) -> List[Dict[str, A
              'to_timestamp': entry[3],
              } for entry in query]
 
+    @staticmethod
+    def hard_reset_assets_list(
+        user_db: 'DBHandler',
+        force: bool = False,
+    ) -> Tuple[bool, str]:
+        """
+        Delete all custom asset entries and repopulate from the last
+        builtin version
+        """
+        root_dir = Path(__file__).resolve().parent.parent
+        builtin_database = root_dir / 'data' / 'global.db'
+        detach_database = 'DETACH DATABASE "clean_db";'
+
+        connection = GlobalDBHandler()._conn
+        cursor = connection.cursor()
+        user_db_cursor = user_db.conn.cursor()
+
+        # Update owned assets
+        user_db.update_owned_assets_in_globaldb()
+
+        try:
+            # First check that the operation can be made. For this create a set with the
+            # difference of the identifiers owned by any user and registered at the globaldb and
+            # the assets identifiers at the packaged db and take the difference. If the difference
+            # is not the empty set the operation is dangerous and the user should be notified.
+            query = cursor.execute('SELECT asset_id from user_owned_assets;')
+            user_ids = {tup[0] for tup in query.fetchall()}
+            # Attach to the clean db packaged with rotki
+            cursor.execute(f'ATTACH DATABASE "{builtin_database}" AS clean_db;')
+            # Get built in identifiers
+            query = cursor.execute('SELECT identifier from clean_db.assets;')
+            shipped_ids = {tup[0] for tup in query.fetchall()}
+            diff_ids = user_ids - shipped_ids
+            if len(diff_ids) != 0 and not force:
+                cursor.execute(detach_database)
+                return False, f'There are assets that can not be deleted: {str(diff_ids)}.'
```

Comment:
Put the assets list in the logs and point the user to the logs in the error message.

I saw that happen and it's indeed unreadable.
```suggestion
                return False, f'There are assets that can not be deleted. Check logs for more details'
```